pulsar-client-go/examples/reader/reader.go

```diff
@@ -0,0 +1,48 @@
+//
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+//
+
+package main
+
+import (
+	"../../pulsar"
+	"fmt"
+	"log"
+)
+
+func main() {
+	client := pulsar.NewClient().
+		ServiceUrl("pulsar://localhost:6650").Build()
+
+	reader, err := client.NewReader().Topic("my-topic").StartFromEarliest().Create()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	for {
+		msg, err := reader.ReadNext()
```

Comment:
Any reason to use `ReadNext` over `Receive()` like with the consumer?

Reply:
This is mostly to differentiate the API semantic difference between the consumer (push) and reader (pull).

Reply:
(the underlying implementation for the delivery is 99% the same, in any case)

Reply:
Fair enough, maybe then just `Next()`?